packages/sol-compiler/src/utils/solc_wrapper.ts

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env node
+// We need the above pragma since this script will be run as a command-line tool.
+
+import { Resolver } from '@0xproject/sol-resolver';
+import { fetchAsync, logUtils } from '@0xproject/utils';
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as requireFromString from 'require-from-string';
+import * as solc from 'solc';
+import * as yargs from 'yargs';
+
+import { constructResolver } from './compiler';
+import { constants } from './constants';
+
+(async () => {
+    const argv = yargs
+        .option('fullSolcVersion', {
+            type: 'string',
+            description: 'full solidity compiler version',
+        })
+        .option('solcBinDir', {
+            type: 'string',
+            description: 'directory holding solc binary',
+        })
+        .option('contractsDir', {
+            type: 'string',
+            description: 'directory holding contracts, for dependency resolution',
+        })
+        .help().argv;
+
+    const compilerBinFilename = path.join(argv.solcBinDir, argv.fullSolcVersion);
+    let solcjs: string;
+    const isCompilerAvailableLocally = fs.existsSync(compilerBinFilename);
+    if (isCompilerAvailableLocally) {
+        solcjs = fs.readFileSync(compilerBinFilename).toString();
+    } else {
+        logUtils.log(`Downloading ${argv.fullSolcVersion}...`);
+        const url = `${constants.BASE_COMPILER_URL}${argv.fullSolcVersion}`;
+        const response = await fetchAsync(url);
+        const SUCCESS_STATUS = 200;
+        if (response.status !== SUCCESS_STATUS) {
+            throw new Error(`Failed to load ${argv.fullSolcVersion}`);
+        }
+        solcjs = await response.text();
+        fs.writeFileSync(compilerBinFilename, solcjs);
+    }
+    const solcInstance = solc.setupMethods(requireFromString(solcjs, compilerBinFilename));
+
+    const stdinBuffer = fs.readFileSync(0);
+    const resolver: Resolver = constructResolver(argv.contractsDir);
+    process.stdout.write(
+        solcInstance.compileStandardWrapper(stdinBuffer.toString(), importPath => {
+            const sourceCodeIfExists = resolver.resolve(importPath);
```

Comment:
We should do an `_.isUndefined` check here. If a variable name ends with `IfExists`, it means it might be undefined. The subsequent indexing into `.source` could fail if it is indeed `undefined`.